### PR TITLE
Remove empty string as default value for VerificationMode 

### PR DESCRIPTION
**Pull request by narph**
- Bug

## What does this PR do?

Original issue https://github.com/elastic/elastic-agent/issues/184
Elastic agent accepts empty string value for `ssl.verification_mode` (which is equivalent to `full`), however Endpoint is logging an error since it is not a supported option, but, it's also setting the verification mode to full, so verification modes match.

The PR is removing empty string as a default/supported verification mode option. Default mode will be `full`.

## Why is it important?

Avoids conflicts with other stakeholders.

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.md`

## Related issues

- Relates https://github.com/elastic/elastic-agent/issues/184

### CODE REVIEW

**Review by fearful-symmetry — APPROVED** [review]
Changes look good to me, although I'm assuming the change itself won't involve any user-facing breaking changes(?)

**Review by belimawr — CHANGES_REQUESTED** [review]
@narph, the PR looks great, but to be on the safe side, I'd like to see test case for when the `verification_mode` is set to an empty string, e.g: test parsing a YAML like this:
```yaml
verification_mode: ""
```

**Comment by narph** [comment]
> @narph, the PR looks great, but to be on the safe side, I'd like to see test case for when the `verification_mode` is set to an empty string, e.g: test parsing a YAML like this:
> 
> ```yaml
> verification_mode: ""
> ```

I've added some tests for this, including one with the scenario you suggested above but I assume by default `verification_mode` is not added in the configuration file instead of having a empty string value. Correct me if I am wrong here.

File: transport/tlscommon/types_test.go [hunk 1]
```diff
@@ -20,6 +20,8 @@ package tlscommon
 import (
 	"testing"
 
+	"github.com/stretchr/testify/assert"
+
```
Comment (belimawr) [inline transport/tlscommon/types_test.go:24]:
```suggestion
```

**Review by belimawr — APPROVED** [review]

**Pull request merged**